roles/k3s/post/tasks/main.yml

```diff
@@ -49,15 +51,15 @@
       condition: --for condition=Ready
     - description: ready replicas of controller
       resource: replicaset
-      selector: component=controller,app=metallb
+      name: '$(kubectl get replicaset -n metallb-system --selector=component=controller,app=metallb --sort-by=.metadata.creationTimestamp -o=jsonpath={.items[-1].metadata.name})'
```

Comment:
@dberardo-com Can you explain what this does?

Reply:
sure, so the problem is that when restarting the daemonSet that controls MetlaLB pods, a new replica set is created. So the list of replicasets will contain more than 1 element from that moment on, and thus this selector will start returning more than one replicaset. But since old replicaset will have "expired" their ready status will be 0 and so this check with "wait" would fail here.

to avoid this it is important to sort the replicasets by date time and always take the latest in the list, that one will be the one with readystate 1

Reply:
the reason why metalLB daemonset needs to be restarted is because otherwise the IP Address Range CRD will not be reloaded and thus the entrypoint's IP not changed

Reply:
okay, I see. IMO, it makes sense to perform the detection of the relevant replicaset in an extra step since the purpose of the extra `kubectl get` is not obvious from just looking at the source (to me, at least).